pkg/cluster/cluster.go

```diff
@@ -1137,3 +1200,29 @@ func (c *Cluster) deletePatroniClusterConfigMaps() error {
 
 	return c.deleteClusterObject(get, deleteConfigMapFn, "configmap")
 }
+
+// Test if two connection pool configuration needs to be synced. For simplicity
+// compare not the actual K8S objects, but the configuration itself and request
+// sync if there is any difference.
+func (c *Cluster) needSyncConnPoolDeployments(oldSpec, newSpec *acidv1.ConnectionPool) (sync bool, reasons []string) {
+	reasons = []string{}
+	sync = false
+
+	changelog, err := diff.Diff(oldSpec, newSpec)
```

Comment:
would the diff function from our [util](https://github.com/zalando/postgres-operator/blob/master/pkg/util/util.go#L69) pkg also work, instead of introducing another dependency?

Reply:
Unfortunately no, already existing `diff` is just for visual comparison of objects. The introduced dependency will find a difference between two arbitrary objects and return a list of what, where and how changed.